runtime/moonbase/Cargo.toml

```diff
@@ -339,9 +342,14 @@ metadata-hash = ["substrate-wasm-builder/metadata-hash"]
 # to make it smaller like logging for example.
 on-chain-release-build = ["metadata-hash", "sp-api/disable-logging"]
 
+# When enabled, the GenesisBuilder API will be supported, GenesisConfig shall be
+# stripped from the final binary
+genesis-builder = ["serde_json"]
+
 runtime-benchmarks = [
 	"cumulus-pallet-parachain-system/runtime-benchmarks",
 	"cumulus-primitives-core/runtime-benchmarks",
+	"genesis-builder",
```

Comment:
We want the `genesis-builder` all nom production builds.

We should rename the feature to `disable-genesis-builder` and reverse the logic to only be disabled when `on-chain-release-build` feature is provided.

Reply:
The problem with that approach is that I can't have `serde_json` as an optional dependency depending on it (or at least I am not sure how to make that work).

What I could do is to add `genesis-builder` as a default feature and change the CI job publishing the runtimes to something like:

```yml
RUNTIME_BUILD_OPTS: "--no-default-features --features=on-chain-release-build"
```

from:
```yml
RUNTIME_BUILD_OPTS: "--features=on-chain-release-build"
```

Q: With the current opts, I think we are using also the `std` feature when publishing them. Is this needed?

Reply:
I think `serde_json` does not need to be an optional dependency, I think it is already included by some nested dependency.

Reply:
Addressed in https://github.com/moonbeam-foundation/moonbeam/pull/3138/commits/c4b071e9381877f419668ce3ebf67fa9002534f8